### ISSUE TITLE
Improvements to site workflow

### ISSUE THREAD

**Issue by stoyanK7** [checkstyle/checkstyle#13206]
In a similar fashion to https://github.com/checkstyle/checkstyle/issues/12991, the workflow can be improved. Lots of logic is in the workflow and can be extracted to a separate script. Both workflows(`site` and  `diff-report`) share an almost identical  last job step(`send_message`). Another issue is that sometimes runners are taken and we have to wait more than 30 minutes inbetween jobs. Placing the whole script under one job will be better - start action -> run script, upload to aws, comment -> free runner for whatever needs it next.

---
Here are things split into chunks with no specific order:
- Move workflow to use only one job instead of 3
- Move each step that contains bash instructions into a `site.sh` script. Simiar to what the description of #12991 describes
    > We should move steps out to a single script file, where we can invoke the script with a single argument command to handle steps, like we do with validation.sh.
- Create a separate script for the identical `send_message` part that `diff-report` and `site`share

**Comment by nrmancuso** [comment]
@stoyanK7 while ideas are fresh in your mind, please create some reasonable list  of clear individual tasks that we can use to break changes into several PRs.

**Comment by YuktiNandwana** [comment]
Hi @stoyanK7 @nrmancuso, I recently worked on improving the CI scripts in .ci/run-link-check-plugin.sh (PR #19098) and would love to help break this down.

To start with the first small PR, I can extract the identical send_message logic from both diff-report and site workflows into a separate dedicated script. Once that is merged, I can pick up the next chunk. Let me know if I can go ahead with this!

**Comment by smita1078** [comment]
On it

**Comment by romani** [comment]
Fix is merged
